### PR TITLE
Add some integration tests

### PR DESCRIPTION
**Pull request by qiwzhang**
* Test Mixer return INTERNAL error,  with fail_close and fail_open policy
* Test adding fault http filter after mixer filter to make sure Check and Report are called.

### CODE REVIEW

**Comment by qiwzhang** [comment]
https://github.com/istio/proxy/issues/409

**Comment by qiwzhang** [comment]
https://github.com/istio/mixerclient/issues/133

File: src/envoy/mixer/integration_test/check_fail_close_test.go [hunk 1]
```diff
@@ -32,7 +32,7 @@ func TestFailClose(t *testing.T) {
 
 	url := fmt.Sprintf("http://localhost:%d/echo", ClientProxyPort)
 
-	tag := "Fail-Open"
+	tag := "Fail-CLOSE"
```
Comment (lizan) [inline src/envoy/mixer/integration_test/check_fail_close_test.go]: `Fail-Close`?

File: src/envoy/mixer/integration_test/setup.go [hunk 2]
```diff
@@ -20,10 +20,11 @@ import (
...
 
 type TestSetup struct {
-	t        *testing.T
-	conf     string
-	stress   bool
-	no_mixer bool
+	t            *testing.T
+	conf         string
+	stress       bool
+	fault_inject bool
```
Comment (lizan) [inline src/envoy/mixer/integration_test/setup.go]: nit: `faultInject` and `noMixer`

File: src/envoy/mixer/integration_test/check_fail_close_test.go (hunk 1 above)
Comment (qiwzhang) [reply src/envoy/mixer/integration_test/check_fail_close_test.go]: Done

File: src/envoy/mixer/integration_test/setup.go (hunk 2 above)
Comment (qiwzhang) [reply src/envoy/mixer/integration_test/setup.go]: Done

**Pull request merged**